git/util.py

```diff
@@ -190,6 +194,10 @@ def _parse_progress_line(self, line):
         # Counting objects: 4, done.
         # Compressing objects:  50% (1/2)   \rCompressing objects: 100% (2/2)   \rCompressing objects: 100% (2/2), done.
         self._cur_line = line
+        if len(self._error_lines) > 0 or self._cur_line.startswith( ('error:', 'fatal:') ):
```

Comment:
Once you see error or fatal the progress either never started or will not continue.

The parsing will never pass an error to update. It only passes data from lines that pattern match against
a "progress" line.